### PR TITLE
added execute, executeAsync and cookies endpoints

### PR DESCRIPTION
**Pull request by endes0**

### CODE REVIEW

**Review by dom96 — CHANGES_REQUESTED** [review]
This is awesome! It would be great to parse the resulting JsonNode as well. As described here: https://github.com/jlipps/simple-wd-spec#execute-script it can either be the return value or an error.

I would make it raise an exception if an error occurs and return a string otherwise. Would you mind making these changes? :)

**Comment by endes0** [comment]
oh, I didn't see that it could return an error. Ok, give me a few days and I will add that and other webdrive commands (cookies).

**Comment by endes0** [comment]
PD: the functions returns a JsonNode because the value can be another type, like int

**Comment by endes0** [comment]
Finally, I was able to do it today.

File: src/webdriver.nim [hunk 1]
```diff
@@ -203,6 +213,126 @@ proc takeScreenshot*(self: Session): string =
...
+    "args": []
+  }
+  for arg in args:
+    obj["args"].elems.add arg
+
+  let resp = self.driver.client.post(reqUrl, $obj)
+  if resp.status != Http200:
+    raise newException(WebDriverException, resp.status)
+
+  let respObj = checkResponse(resp.body)
```
Comment (dom96) [inline src/webdriver.nim:226]: I believe `checkResponse` will raise an exception so the line below will have no effect.

**Review by dom96 — APPROVED** [review]
Other than that one small thing, looks good. Please fix and I'll merge.

**Comment by dvn0** [comment]
Is this ready to be merged @dom96?

File: src/webdriver.nim (hunk 1 above)
Comment (dvn0) [reply src/webdriver.nim:226]: Looks like this commit was the attempt to resolve this, no? https://github.com/dom96/webdriver/pull/7/commits/1f0b47f4eb59fc285d8afcff736afd175621851f

Comment (dom96) [reply src/webdriver.nim:226]: Not sure I understand, this is still a problem. When the JSON has `error` in it, then `checkResponse` will raise because there will be no `value` key (https://github.com/dom96/webdriver/blob/1f0b47f4eb59fc285d8afcff736afd175621851f/src/webdriver.nim#L46). That's my assumption, maybe I'm wrong, but if not then the code below will never run.

**Review by dom96 — CHANGES_REQUESTED** [review]

File: src/webdriver.nim (hunk 1 above)
Comment (endes0) [reply src/webdriver.nim:226]: Error is inside the value key (https://github.com/jlipps/simple-wd-spec#execute-script). I forgot to put `respObj["value"]`.

File: src/webdriver.nim [hunk 2]
```diff
@@ -203,6 +213,106 @@ proc takeScreenshot*(self: Session): string =
...
+    if cookie.hasKey("path"):
+      final.path = some(cookie["path"].getStr)
+    if cookie.hasKey("domain"):
+      final.domain = some(cookie["domain"].getStr)
+    if cookie.hasKey("secure"):
+      final.secure = some(cookie["secure"].getBool)
+    if cookie.hasKey("httpOnly"):
+      final.httpOnly = some(cookie["httpOnly"].getBool)
+    if cookie.hasKey("expiry"):
+      final.expiry = some(cookie["expiry"].getBiggestInt)
```
Comment (dom96) [inline src/webdriver.nim:305]: There is some repetition here that could be refactored. Do so in a separate PR if you're up for it.

**Review by dom96 — APPROVED** [review]
Thanks for the fixes. Looks good to merge.
